pkg/apis/openebs.io/upgrade/v1alpha1/upgradetask.go

```diff
@@ -0,0 +1,195 @@
+/*
+Copyright 2019 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +resource:path=upgradetask
+// +k8s:openapi-gen=true
+
+// UpgradeTask represents an upgrade task
+type UpgradeTask struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	// Spec i.e. specifications of the UpgradeTask
+	Spec UpgradeTaskSpec `json:"spec" protobuf:"bytes,2,opt,name=spec"`
+	// Status of UpgradeTask
+	Status UpgradeTaskStatus `json:"status" protobuf:"bytes,3,opt,name=status"`
+}
+
+// UpgradeTaskSpec is the properties of an upgrade task
+type UpgradeTaskSpec struct {
+	// FromVersion is the current version of the resource.
+	FromVersion string `json:"fromVersion" protobuf:"bytes,1,name=fromVersion"`
+	// ToVersion is the upgraded version of the resource. It should be same
+	// as the version of control plane components version.
+	ToVersion string `json:"toVersion" protobuf:"bytes,2,name=toVersion"`
+	// Flags contains the optional flags that can be passed during upgrade.
+	Flags *Flags `json:"flags,omitempty" protobuf:"bytes,3,name=flags"`
```

Comment:
Flags sounds like boolean.. can we change this to something like `Config`.. please check with others also before modifying code

Reply:
Can we rename this to Options or Config

Reply:
Renamed it to options